gdal/src/main/scala/geotrellis/contrib/vlm/gdal/GDALDataPath.scala

```diff
@@ -41,261 +42,75 @@ import java.net.MalformedURLException
  */
 case class GDALDataPath(
   path: String,
-  compressedFileDelimiter: String = "!"
+  compressedFileDelimiter: Option[String] = "!".some,
+  percentEncoder: PercentEncoder = PercentEncoder(PATH_CHARS_TO_ENCODE ++ Set('%', '?', '#'))
```

Comment:
Are these truly parameters? It seems odd to have them here in the `case class` and also I'm not sure they ever change. C/D?

Reply:
@echeipesh without these params some weird S3 paths may not work; actually I don't know what is the entire list of symbols we want to encode, that's why I decided to make it a parameter, that can be a parameter of the parsing function that would be in a companion object (we talked about it earlier today).

Reply:
Naive question, should we just use [`URLDecoder`](https://docs.oracle.com/javase/7/docs/api/java/net/URLDecoder.html) because presumably it handles all the cases?

Reply:
@echeipesh will give it a try

Reply:
@echeipesh 

```
------PercentEncoder---------
/Users/me/subversions/git/github/locationtech/geotrellis-contrib/gdal/target/scala-2.11/test-classes/img/aspect-tiled.tif
------------------------------
%2FUsers%2Fme%2Fsubversions%2Fgit%2Fgithub%2Flocationtech%2Fgeotrellis-contrib%2Fgdal%2Ftarget%2Fscala-2.11%2Ftest-classes%2Fimg%2Faspect-tiled.tif
------URLEncoder--------------
```